app/Http/Controllers/SportsController.php

```diff
@@ -2,31 +2,76 @@
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\SportsStoreRequest;
+use App\Models\Medal;
 use App\Models\Sport;
 use App\Models\Country;
 use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Str;
 
 class SportsController extends Controller
 {
     public function create()
     {
         $sports = Sport::all();
         $countries = Country::all();
+        $medal_names = Medal::NAMES;
 
-        return view('sports.create', compact('sports', 'countries'));
+        return view('sports.create', compact('sports', 'countries', 'medal_names'));
     }
 
-    public function store(Request $request)
+    public function store(SportsStoreRequest $request)
     {
-        // Add your code here
+        // Added my own code here
+        // return $request->all();
+
+        DB::transaction(function () use ($request) {
+            foreach (Medal::NAMES as $medal_name) {
+
+                foreach ($request[Str::plural($medal_name)] as $sport_id => $country_id) {
+                    $country_query = Sport::find($sport_id)
```

Comment:
@shibbirweb doing `::find()` inside of foreach loop means 1 extra query to the database each time. It's better to have all sports queried upfront, and then work with a collection.

Reply:
@PovilasKorop Thanks a lot...